src/python/gudhi/alpha_complex.pyx

```diff
@@ -28,66 +28,72 @@ __license__ = "GPL v3"
 
 # AlphaComplex python interface
 cdef class AlphaComplex:
-    """AlphaComplex is a simplicial complex constructed from the finite cells
-    of a Delaunay Triangulation.
+    """AlphaComplex is a simplicial complex constructed from the finite cells of a Delaunay Triangulation.
 
-    The filtration value of each simplex is computed as the square of the
-    circumradius of the simplex if the circumsphere is empty (the simplex is
-    then said to be Gabriel), and as the minimum of the filtration values of
-    the codimension 1 cofaces that make it not Gabriel otherwise.
+    The filtration value of each simplex is computed as the square of the circumradius of the simplex if the
+    circumsphere is empty (the simplex is then said to be Gabriel), and as the minimum of the filtration values of the
+    codimension 1 cofaces that make it not Gabriel otherwise.
 
-    All simplices that have a filtration value strictly greater than a given
-    alpha squared value are not inserted into the complex.
+    All simplices that have a filtration value strictly greater than a given alpha squared value are not inserted into
+    the complex.
 
     .. note::
 
-        When Alpha_complex is constructed with an infinite value of alpha, the
-        complex is a Delaunay complex.
-
+        When Alpha_complex is constructed with an infinite value of alpha, the complex is a Delaunay complex.
     """
 
     cdef Alpha_complex_interface * this_ptr
 
     # Fake constructor that does nothing but documenting the constructor
-    def __init__(self, points=None, off_file='', precision='safe'):
+    def __init__(self, points=[], off_file='', weights=None, precision='safe'):
         """AlphaComplex constructor.
 
         :param points: A list of points in d-Dimension.
-        :type points: list of list of double
-
-        Or
+        :type points: Iterable[Iterable[float]]
 
-        :param off_file: An OFF file style name.
+        :param off_file: **[deprecated]** An `OFF file style <fileformats.html#off-file-format>`_ name.
+            If an `off_file` is given with `points` as arguments, only points from the file are taken into account.
         :type off_file: string
 
+        :param weights: A list of weights. If set, the number of weights must correspond to the number of points.
+        :type weights: Iterable[float]
+
         :param precision: Alpha complex precision can be 'fast', 'safe' or 'exact'. Default is 'safe'.
         :type precision: string
+
+        :raises FileNotFoundError: **[deprecated]** If `off_file` is set but not found.
+        :raises ValueError: In case of inconsistency between the number of points and weights.
         """
 
     # The real cython constructor
-    def __cinit__(self, points = None, off_file = '', precision = 'safe'):
+    def __cinit__(self, points = [], off_file = '', weights=None, precision = 'safe'):
```

Comment:
Does this mean I can construct `AlphaComplex()`? Well, I guess we can remove the default value of `points` when we eventually remove `off_file`, so in the mean time this default value is not that important.

Reply:
None was used to detect unweighted alpha complex. An empty list works as well and simplifies the code.
Done on c1cf7fe3 and c1cf7fe3
And yes, you can construct `AlphaComplex()`
```python
import gudhi as gd
ac = gd.AlphaComplex(precision='fast') # or 'exact' or 'safe'
st = ac.create_simplex_tree()
st.num_vertices()
## 0
```

Reply:
My comment was about `points`, not `weight`.  I think ultimately we want `points` without a default value, and `weight=None`. I was saying that while I prefer `points=None` to `points=[]`, since we will remove this default value eventually, I am ok with the suboptimal `[]` for now. Changing `None` to to `[]` looks wrong and reverts a change asked in a previous comment.

Reply:
Ok, sorry about that I completely missed your point and I rolled back on cee14f4.
The problem is, if `points` has no default value, as `off_file` is deprecated and not yet removed, users will have to set also `points` when using `off_file` parameter... That will not be very intuitive, no ?